### DISCUSSION TITLE
EnvironmentObject에서 Saving 객체를 사용할 것인가

### DISCUSSION THREAD

**Discussion by unuhqueen** [DeveloperAcademy-POSTECH/toGather#70] [General]
현재 environmentObject 상에서는 다음과 같이 기존의 Model 속 Saving 클래스의 인스턴스를 만들어 Firebase 업로드 용으로 사용 중입니다.

<img width="929" alt="스크린샷 2022-06-16 오전 9 14 10" src="https://user-images.githubusercontent.com/49133245/173962784-563eacf2-7bd6-4640-a55a-fbeed59fd243.png">

그렇다 보니 다음과 같이 Saving 클래스의 프로퍼티들은 반드시 var로 선언해야 하는 상황이 발생합니다.

<img width="496" alt="스크린샷 2022-06-16 오전 9 14 24" src="https://user-images.githubusercontent.com/49133245/173962865-8fc5ffdf-d575-49ac-8af0-1aec36d326ae.png">

그래서 만약 saving 업로드용 객체를 따로 만들면 기존의 saving 인스턴스는 let으로 선언하지 않아도 되지 않나 하는 의견이 있어 discussion 안건으로 부쳐봅니다.

**Comment by unuhqueen** [comment]
이걸 쓰다보니 생각났는데 만약에 저축 목표가 바뀐다고 해도 let을 사용할 수 있는건가요?

**Reply by yeongwooCho** [reply]
let 으로 변경하는 게 목적이라면, 각 페이지 마다 새로운 인스턴스를 생성하고 전달하는 방식은 어떤가요?

**Comment by yeongwooCho** [comment]
그럼 데이터 형식은 같고 var에서 let으로 선언된 모델만 하나 더 생기는 게 아닐까요?

**Comment by limhyoseok** [comment]
관건은 온보딩 시작하자마자 Saving 객체를 초기화할 것인지 아니면 온보딩이 끝나고 초기화할 것인지인 것 같습니다.
(Saving 구조체는 MainView에서 사용할 데이터를 담고 있음)

## 논의할 점
- var로 선언된 goalProduct, goalWeeks... 등등은 변하지 않을 값인데 초기화에 사용한 임의값을 수정하기 위한 용도로 var을 사용?
- 초기에 Saving 구조체를 초기화하기 위해 임의의 값을 넣어줄 경우 computated property 계산 과정에서 문제가 생길 수 있다.
아래는 Saving 구조체에 있는 progressPercent인데 goalWeeks 값이 0으로 나룰 경우 문제가 발생
<img width="783" alt="image" src="https://user-images.githubusercontent.com/59302419/173963765-e2474e91-b053-4a91-936e-604de38414d4.png">

## 제안 
온보딩 상에서 전달되는 데이터는 5가지로 많은 양이 아니기 때문에 그냥 딕셔너리 같은 객체에 임시로 정보를 담은 다음에 온보딩이 끝나고 Saving을 초기화하는 것은 어떤가요? 또한 온보딩 내에서도 해당 데이터는 변하지 않기 때문에 state property도 필요없을 것 같고 뷰 넘어갈 때마다 전달하는 방식도 가능할 것 같습니다.

<img width="1063" alt="image" src="https://user-images.githubusercontent.com/59302419/173964259-0ad50497-4ab3-45e2-87f3-54b3ab9ecf56.png">
